src/endpoint.rs

```diff
@@ -351,18 +349,13 @@ impl Endpoint {
             Some(Err(error)) => return Err(error.into()),
             None => {}
         }
+        let final_conn = retry(ExponentialBackoff::default(), || async {
```

Comment:
We could put this and the `create_new_connection` retry into `attempt_connection` instead, since it's called from both (and only by those). Since it's just changing what was there before though that could be a future thing.